src/components/dashjs-api-control-element/dashjs-api-control-element.tsx

```diff
@@ -14,57 +14,41 @@ export class DashjsAPIControlElement {
 
-  @State() functionValue : any;
+  @State() functionValue : any = "";
 
   setValue(val) {
     this.functionValue = val;
   }
 
   componentWillLoad() {
-    if (this.options != undefined) {
-      this.control = [
-        <ion-select placeholder="Select One" interface="popover" value={this.defaultValue} onIonChange={change => this.valueChanged.emit(change.detail.value)}>
-          {this.options.map(val => (
-            <ion-select-option value={val}>{val}</ion-select-option>
-          ))}
-        </ion-select>,
-        // <ion-label position="floating">{this.name}</ion-label>,
-        // <ion-select placeholder="Select One" interface="popover" value={this.defaultValue} onIonChange={change => this.valueChanged.emit(change.detail.value)}>
-        //   {this.options.map(val => (
-        //     <ion-select-option value={val}>{val}</ion-select-option>
-        //   ))}
-        // </ion-select>,
-      ];
-    } else if (this.param.length < 1) {
-      this.control = [
-        <ion-item lines="none"><ion-button shape="round" size="small" onClick={ () => this.valueChanged.emit("")}>Call</ion-button></ion-item>
-      ];
-    } else if (this.param.length === 1) {
-      switch (this.param[0].type) {
-        case "string":
-          this.control = [
-          <ion-item lines="none"><ion-input debounce={300} value={this.defaultValue} onIonChange={change => this.valueChanged.emit(change.detail.value)}></ion-input>
-          <div class="gap"></div><ion-button size="small" class="left_gap" shape="round" onClick={ () => this.valueChanged.emit("")}>Call</ion-button></ion-item>
-        ];
-          break;
-        case "number":
-          this.control = [
-            <ion-item lines="none"><ion-input debounce={300} type="number" value={this.defaultValue} onIonChange={change => this.setValue(change.detail.value)}></ion-input>
-            <div class="gap"></div><ion-button size="small" shape="round" onClick={ () => this.valueChanged.emit(this.functionValue)}>Call</ion-button></ion-item>
-            // <ion-range min={20} max={80} step={2}>
-            //   <ion-icon size="small" slot="start" name="sunny"></ion-icon>
-            //   <ion-icon slot="end" name="sunny"></ion-icon>
-            // </ion-range>
-          ];
-          break;
-        case "boolean":
-          this.control = [
-          <ion-item lines="none"><ion-toggle value={this.defaultValue} onIonChange={change => this.setValue(change.detail.checked)}></ion-toggle>
-          <ion-button size="small" shape="round" onClick={ () => this.valueChanged.emit(this.functionValue)}>Call</ion-button></ion-item>
-          ];
-          break;
+    if (this.param.length > 0) {
+      for(var i = 0; i <this.param.length; i++) {
+        switch (this.param[i].type) {
+          case "string":
+            this.control.push(<ion-input class="input-border" debounce={300} value={this.defaultValue} onIonChange={change => this.setValue(change.detail.value)}></ion-input>);
```

Comment:
is set value really working if there are multiple input controls? It all setting it to the same `functionValue`?